packages/remix-dev/cli/create.ts

```diff
@@ -195,8 +195,8 @@ export async function createApp({
     !useTypeScript &&
     fse.existsSync(path.join(projectDir, "tsconfig.json"))
   ) {
-    let spinner = ora("Converting template to JavaScript…").start();
-    await convertTemplateToJavaScript(projectDir);
+    let spinner = ora("Migrating template to JavaScript…").start();
+    await convertToJavaScript({ flags: {}, projectDir });
```

Comment:
Can you make the `flags` parameter optional? Also might be good to rename to `options` instead of `flags` since "flags" is more of a CLI term and `convertToJavascript` can be called programmatically like we do here.

Also, I think a nicer function signature for `convertToJavascript` might be:

```ts
function convertToJavascript(projectDir: string, options?: { /* relevant options corresponding to flags */} = {})
```

So that calling it here would be:

```suggestion
    await convertToJavaScript(projectDir);
```

Reply:
This would mean we'll have to refactor the `MigrationFunction` signature as well.
Do we want this for all of the migrations then?
https://github.com/remix-run/remix/blob/f3691d51027b93caa3fd2cdfe146d7b62a6eb8f2/packages/remix-dev/cli/migrate/types.ts#L3-L6

Also: do you want the `Flags` type to be renamed to `Options` as well?
If so, we would need to rename all `flags` args to `options` as well, which isn't always suited imo.

In `packages/remix-dev/cli/migrate/jscodeshift.ts` for instance, it's actually representing JSCodeShift flags
https://github.com/remix-run/remix/blob/f3691d51027b93caa3fd2cdfe146d7b62a6eb8f2/packages/remix-dev/cli/migrate/jscodeshift.ts#L15-L20

In `packages/remix-dev/cli/migrate/resolveInput.ts` it's representing CLI flags
https://github.com/remix-run/remix/blob/f3691d51027b93caa3fd2cdfe146d7b62a6eb8f2/packages/remix-dev/cli/migrate/resolveInput.ts#L38-L54

Reply:
Yeah, I didn't have an issue with `flags` because that's the JS API for jscodeshift. I don't mind switching it to `options` though. Maybe any flags that go to `jscodeshift` would be on a sub-option called "flags" and all our own options would go as properties on the options object?

I do like the `projectDir` as the first argument though.